unstructured/ingest/connector/mongodb.py

```diff
@@ -78,39 +88,203 @@ def to_dict(
+                f.write(concatenated_values)
+
+            self.ingest_docs.append(ingest_doc)
+
+
+@dataclass
+class MongoDBSourceConnector(SourceConnectorCleanupMixin, BaseSourceConnector):
+    connector_config: SimpleMongoDBConfig
     _client: t.Optional["MongoClient"] = field(init=False, default=None)
 
+    @property
+    def client(self) -> "MongoClient":
+        if self._client is None:
+            self._client = self.connector_config.generate_client()
+        return self._client
+
+    def check_connection(self):
+        try:
+            self.client.admin.command("ping")
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise DestinationConnectionError(f"failed to validate connection: {e}")
+
+    def initialize(self):
+        _ = self.client
+
     @requires_dependencies(["pymongo"], extras="mongodb")
-    def generate_client(self) -> "MongoClient":
-        from pymongo import MongoClient
-        from pymongo.server_api import ServerApi
+    def _get_doc_ids(self):
+        """Fetches all document ids in a collection."""
+        collection = self.connector_config.get_collection(self.client)
+        return [str(x) for x in collection.distinct("_id")]
 
-        if self.connector_config.uri:
-            return MongoClient(
-                self.connector_config.uri, server_api=ServerApi(version=SERVER_API_VERSION)
+    def get_ingest_docs(self):
+        """Fetches all documents in an index, using ids that are fetched with _get_doc_ids"""
+        ids = self._get_doc_ids()
+        id_batches = [
+            ids[
+                i
+                * self.connector_config.batch_size : (i + 1)  # noqa
+                * self.connector_config.batch_size
+            ]
+            for i in range(
+                (len(ids) + self.connector_config.batch_size - 1)
+                // self.connector_config.batch_size
             )
-        else:
-            return MongoClient(
-                host=self.connector_config.host,
-                port=self.connector_config.port,
-                server_api=ServerApi(version=SERVER_API_VERSION),
+        ]
+
+        return [
+            MongoDBIngestDocBatch(
+                connector_config=self.connector_config,
+                processor_config=self.processor_config,
+                read_config=self.read_config,
+                list_of_ids=batched_ids,
             )
+            for batched_ids in id_batches
+        ]
+
+
+@dataclass
+class MongoDBDestinationConnector(BaseDestinationConnector):
+    connector_config: SimpleMongoDBConfig
+    _client: t.Optional["MongoClient"] = field(init=False, default=None)
 
     @property
     def client(self) -> "MongoClient":
         if self._client is None:
-            self._client = self.generate_client()
+            self._client = self.connector_config.generate_client()
         return self._client
 
```

Comment:
while in here, please define a to_dict method and in it remove this _client since it won't be serializable. Hereis a reference PR to where this has been done for another connector

Reply:
Added the to_dict. Didn't remove _client.

Reply:
You need to remove the client in the to_dict method. See that pr for reference

Reply:
@ryannikolaidis Sorry. Maybe I didn't communicate well. The elasticsearch version sets the `_client` attribute to None. I did exactly the same thing. Is that what you mean by remove? Or possibly I'm not understanding.

Reply:
@potter-potter yep, that's exactly what I meant. thanks!